clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy

```diff
@@ -75,11 +89,22 @@ class AzureResourceManagerClient extends AzureBaseClient {
     }
   }
 
-  public boolean resourceGroupExists(AzureCredentials creds, String resourceGroupName) {
-    return this.getResourceManagementClient(creds).getResourceGroupsOperations().checkExistence(resourceGroupName).isExists()
+  ArrayList<ResourceGroup> getResourcesGroupsForApp(AzureCredentials creds, String applicationName) {
+    ResourceGroupListParameters parameters = new ResourceGroupListParameters()
+    parameters.setTagName("filter")
+    parameters.setTagValue(applicationName)
+    return this.getResourceManagementClient(creds).getResourceGroupsOperations().list(parameters).resourceGroups
+  }
+
+  ArrayList<ResourceGroupExtended> getAllResourceGroups(AzureCredentials creds) {
+    this.getResourceManagementClient(creds).getResourceGroupsOperations().list(null).getResourceGroups()
+  }
+
+  boolean resourceGroupExists(AzureCredentials creds, String resourceGroupName) {
+    this.getResourceManagementClient(creds).getResourceGroupsOperations().checkExistence(resourceGroupName).isExists()
   }
 
-  public void healthCheck(AzureCredentials creds) throws Exception {
+  void healthCheck(AzureCredentials creds) throws Exception {
```

Comment:
Minor, and I realize this was pre-existing, but `throws Exception` is generally something to be avoided and mostly unnecessary in Groovy.

If there are concrete exception types, feel free to list them explicitly.